compiler/codegen/OMRAheadOfTimeCompile.cpp

```diff
@@ -115,6 +116,31 @@ void OMR::AheadOfTimeCompile::traceRelocationOffsets(uint8_t *&cursor, int32_t o
       }
    }
 
+
+#ifndef J9_PROJECT_SPECIFIC
```

Comment:
My focus on the PR currently is to get rid of the #define guards. My approach is to bring similar to OpenJ9, but not identical AOT-related code into OMR, introducing some code duplication. I have refactored the classes in my PR, for symbols to be different and not interfere with OpenJ9 classes that are not planned to be an extension in openJ9 (as of now). However, the AheadOfTimeCompile was designed to be extensible class, so I have done some changes  in OMR so that J9AheadOfTimeCompile is an extension class of an OMRAheradOfTimeCompile. I have ran tests that showed some unresolved symbols. Turns out, OpenJ9 does not fully depend on CMake and the OMR version of AheadOfTimeCompile was not included into OpenJ9 build. Changin OpenJ9 makefiles introduced new symbol errors, so more have to be included in OpenJ9 build. I do not have enough expertise, whether I should include my files in OpenJ9 build procedure and open the corresponding PR to OpenJ9, or shall I rename the functions that OpenJ9 needs for symbol resolution purposes? The first option will change the way OpenJ9 builds and will require a PR to OpenJ9 repository, the second option will make OMR having more code that is not used by OpenJ9 at this stage, but this can be changed later.

Reply:
Perhaps the best way to deal with this right now is, instead of having `#ifndef J9_PROJECT_SPECIFIC` guard it with a `#define` that you'll have to enable in the OMR build system. OpenJ9 will not define this macro. 

Trying to reconcile OpenJ9 builds might be too much to do in this PR.

Reply:
I agree with @dsouzai 's suggestion. Making a change like this in one shot is unlikely to be successful. It will mean a progression of PRs will be needed. Maybe something like `OMR_RELOCATION_RUNTIME` .

Not a big fan of the word "Default" in a query. Maybe `useOMRRelocationFramework()` (also not great) ... suggest you introduce such a thing as an independent PR in OMR (to provide the default `return true` impl) then OpenJ9 (to override default impl with `return false`) then you can restructure this PR using that query.

Reply:
Hello, thank you for your comments @dsouzai, @mstoodle ! That actually looks like a way simpler solution than I had in mind. I will roll back my changes for a while, and follow the suggested path. To my understanding, the query should be implemented in place of ```compileRelocatableCode()```, while the define guards can be used to avoid build-time errors. How does that suggestion look to you?

Reply:
This concern was resolved in later comments